flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/DynamicKafkaRecordSerializationSchemaTest.java

```diff
@@ -0,0 +1,147 @@
+package org.apache.flink.streaming.connectors.kafka.table;
+
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.connector.kafka.sink.KafkaRecordSerializationSchema;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.util.TestLogger;
+
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/** Tests for {@link DynamicKafkaRecordSerializationSchema}. */
+public class DynamicKafkaRecordSerializationSchemaTest extends TestLogger {
+    private static final List<String> TOPICS = Arrays.asList("topic1;topic2".split(";"));
+    private static final String TOPIC = "topic";
+    private static final Pattern TOPIC_PATTERN = Pattern.compile("topic*");
+
+    @ParameterizedTest
+    @MethodSource("provideTopicMetadataTestParameters")
+    public void testTopicMetadata(
+            List<String> topics, Pattern topicPattern, String rowTopic, String expectedTopic) {
+        GenericRowData rowData = createRowData(rowTopic);
+        DynamicKafkaRecordSerializationSchema schema = createSchema(topics, topicPattern);
+        KafkaRecordSerializationSchema.KafkaSinkContext context = createContext();
+
+        // Call serialize method
+        ProducerRecord<byte[], byte[]> record = schema.serialize(rowData, context, null);
+
+        // Assert the returned ProducerRecord is routed to the correct topic
+        assertEquals(record.topic(), expectedTopic);
```

Comment:
I'm not sure what the default is now, but you mix assertj and junit5 assertions in this test. How about using just assertj?

`assertThat(record.topic()).isEqualTo(expectedTopic)`